impl/src/main/java/io/kroxylicious/testing/kafka/common/KeytoolCertificateGenerator.java

```diff
@@ -77,17 +120,31 @@ public void generateSelfSignedCertificateEntry(String email, String domain, Stri
         commandParameters.addAll(List.of("-keysize", "2048"));
         commandParameters.addAll(List.of("-sigalg", "SHA256withRSA"));
         commandParameters.addAll(List.of("-storetype", "JKS"));
-        commandParameters.addAll(List.of("-keystore", getCertLocation()));
+        commandParameters.addAll(List.of("-keystore", getKeyStoreLocation()));
         commandParameters.addAll(List.of("-storepass", getPassword()));
         commandParameters.addAll(List.of("-keypass", getPassword()));
         commandParameters.addAll(
                 List.of("-dname", getDomainName(email, domain, organizationUnit, organization, city, state, country)));
         commandParameters.addAll(List.of("-validity", "365"));
         commandParameters.addAll(List.of("-deststoretype", "pkcs12"));
-        commandParameters.addAll(List.of("-storetype", "JKS"));
         if (canGenerateWildcardSAN() && !isWildcardDomain(domain)) {
             commandParameters.addAll(getSAN(domain));
         }
+        runCommand(commandParameters);
+
+        createCrtFileToImport();
+    }
+
+    private void createCrtFileToImport() throws IOException {
+        final List<String> commandParameters = new ArrayList<>(List.of("openssl", "pkcs12"));
+        commandParameters.addAll(List.of("-in", getKeyStoreLocation()));
+        commandParameters.addAll(List.of("-passin", "pass:" + getPassword()));
+        commandParameters.add("-nokeys");
+        commandParameters.addAll(List.of("-out", certFilePath.toAbsolutePath().toString()));
+        runCommand(commandParameters);
+    }
+
+    private void runCommand(List<String> commandParameters) throws IOException {
```

Comment:
Oh yikes we can't use key tool via a native API? We have to invoke it as an external process.

Reply:
I have investigated this and the native API is not public any more since Java 9 I believe, I would need to look for that info again to be sure from which version it is not available.